src/test/java/uk/nhs/digital/nhsconnect/lab/results/inbound/queue/InboundQueueServiceTest.java

```diff
@@ -38,42 +46,107 @@
     @Spy
     private TimestampService timestampService;
 
-    @Mock
-    private ConversationIdService conversationIdService;
-
     @Captor
     private ArgumentCaptor<MessageCreator> jmsMessageCreatorCaptor;
 
+    @InjectMocks
+    private InboundQueueService inboundQueueService;
+
     @Mock
     private JmsTemplate jmsTemplate;
 
-    @InjectMocks
-    private InboundQueueService inboundQueueService;
+    @Mock
+    private ConversationIdService conversationIdService;
+
+    @Mock
+    private Message message;
+
+    @Mock
+    private InboundQueueConsumerService inboundQueueConsumerService;
+
+    @Test
+    void receiveInboundMessageIsHandledByInboundQueueConsumerService() throws Exception {
+        when(message.getStringProperty(JmsHeaders.CONVERSATION_ID)).thenReturn(CONVERSATION_ID);
+        when(message.getBody(String.class)).thenReturn("{\"workflowId\":\"LAB_RESULTS_REG\"}");
+
+        inboundQueueService.receive(message);
+
+        verify(conversationIdService).applyConversationId(CONVERSATION_ID);
+
+        final MeshMessage expectedMeshMessage = new MeshMessage();
+        expectedMeshMessage.setWorkflowId(WorkflowId.REGISTRATION);
+        verify(inboundQueueConsumerService).handle(expectedMeshMessage);
+
+        verify(message).acknowledge();
+        verify(conversationIdService).resetConversationId();
+    }
+
+    @Test
+    void receiveInboundMessageWithInvalidWorkflowIdThrowsException() throws Exception {
+        when(message.getStringProperty(JmsHeaders.CONVERSATION_ID)).thenReturn(CONVERSATION_ID);
+        when(message.getBody(String.class)).thenReturn("{\"workflowId\":\"INVALID\"}");
+
+        assertThrows(Exception.class, () -> inboundQueueService.receive(message));
+
+        verify(conversationIdService).applyConversationId(CONVERSATION_ID);
+        verify(message, times(0)).acknowledge();
```

Comment:
We can use `never()` instead of `times(0)` to make this more immediately obvious to the reader.

And elsewhere in this PR.

Reply:
Sure 👍